crypto/rand/rand_lcl.h

```diff
@@ -17,28 +17,25 @@
 # include <openssl/ec.h>
 # include "internal/rand.h"
 
-/*
- * Amount of randomness (in bytes) we want for initial seeding.
- * This is based on the fact that we use AES-128 as the CRBG, and
- * that we use the derivation function.  If either of those changes,
- * (see rand_init() in rand_lib.c), change this.
- */
-# define RANDOMNESS_NEEDED              16
 
 /* How many times to read the TSC as a randomness source. */
 # define TSC_READ_COUNT                 4
 
-/* Maximum amount of randomness to hold in RAND_BYTES_BUFFER. */
-# define MAX_RANDOMNESS_HELD            (4 * RANDOMNESS_NEEDED)
-
 /* Maximum count allowed in reseeding */
 # define MAX_RESEED                     (1 << 24)
 
-/* How often we call RAND_poll() in drbg_entropy_from_system */
-# define RAND_POLL_RETRIES 8
+/* Max size of additional input and personalization string. A reasonably large value ;-) */
+# define DRBG_MAX_LENGTH                1024
 
-/* Max size of entropy, addin, etc. Larger than any reasonable value */
-# define DRBG_MAX_LENGTH                0x7ffffff0
+/*
+ * The quotient between max_{entropy,nonce}len and min_{entropy,nonce}len
+ * 
+ * The current factor is large enough that the RAND_POOL can store a
+ * random input which has a lousy entropy rate of 0.25 bits per byte.
+ * This input will be sent through the derivation function which 'compresses'
+ * the low quality input into a high quality output
+ */
+# define DRBG_MINMAX_FACTOR              32
```

Comment:
Could the factor be made larger?  Allowing down to _0.1_ bits per byte perhaps?  There is no harm in making the buffer bigger, a higher quality source will always terminate early I believe.

I have some test data from `/proc/net/dev` on an embedded ARM processor that gave _0.14_ bits per byte when assessed by the NIST SP 800-90B second draft estimates.  Small processors like these are notorious for having difficulty finding entropy, and a number of poor sources are often the only recourse.

I actually have far worse data sets, one at _0.033_ and one at _0.010_ bit per byte but they've problems passing some of the second draft tests.  My worst ever collected is an impressive _0.0022_ bits per byte (using the first draft SP 800-90B tests), I didn't bother running the second draft tests for this one.

Reply:
Of course. We're not talking about a huge amount of memory here. It was just some initial value and I didn't want to be too wasteful.

Reply:
Cool :)

Reply:
I like powers of two, so how about taking 128? 
```C
# define DRBG_MINMAX_FACTOR              128
```
This would correspond to an entropy ratio of 0.0625 bits per byte. One caveat however is that the resolution of `RAND_POOL_bytes_needed()` only goes down to 1 bit per byte:
```
int RAND_POOL_bytes_needed(RAND_POOL *pool, unsigned int entropy_per_byte)
```
I would object to replacing the integer by a double, or using something like bits-per-hundert-bytes instead. However, `RAND_POOL_bytes_needed()` is only a convenience method and you can always do the easy calculations by yourself, using the other accessors.

Reply:
This sounds reasonable, I don't mind if it is a power or two or not.  I agree, avoid doubles unless we go all in using them (which isn't going to happen).